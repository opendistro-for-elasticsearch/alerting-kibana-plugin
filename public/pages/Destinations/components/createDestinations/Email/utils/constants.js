/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

export const METHOD_TYPE = {
  NONE: 'none',
  SSL: 'ssl',
  TLS: 'starttls',
};

export const STATE = {
  NO_OP: 'no_op',
  UPDATED: 'updated',
  CREATED: 'created',
};

export const FORMIK_INITIAL_SENDER_VALUES = {
  name: '',
  email: '',
  host: '',
  port: '',
  method: METHOD_TYPE.NONE,
  state: STATE.CREATED,
};

export const FORMIK_INITIAL_EMAIL_GROUP_VALUES = {
  name: '',
  emails: [],
  state: STATE.CREATED,
};
