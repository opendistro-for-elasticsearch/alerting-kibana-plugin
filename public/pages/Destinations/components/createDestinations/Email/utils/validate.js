/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import _ from 'lodash';

export const validateSenderName = (senders) => (value) => {
  if (!value) return 'Required';

  if (!/^[A-Z0-9_]+$/i.test(value)) return 'Invalid sender name';

  const matches = senders.filter((sender) => sender.name === value);
  if (matches.length > 1) return 'Sender name is already being used';
};

export const validateEmailGroupName = (emailGroups) => (value) => {
  if (!value) return 'Required';

  if (!/^[A-Z0-9_-]+$/i.test(value)) {
    return 'Invalid email group name';
  }

  const matches = emailGroups.filter((emailGroup) => emailGroup.name === value);
  if (matches.length > 1) return 'Email group name is already being used';
};

export const validateEmailGroupEmails = (options) => {
  if (_.isEmpty(options)) return 'Must specify an email';

  const invalidEmails = options
    .map((option) => option.label)
    .filter((email) => !isValidEmail(email));
  if (invalidEmails.length > 0) {
    return `Invalid emails: ${invalidEmails.join(', ')}`;
  }
};

export const validateEmail = (value) => {
  if (!value) return 'Required';

  if (!isValidEmail(value)) {
    return 'Invalid email address';
  }
};

export const validateHost = (value) => {
  // SMTP hosts are usually of the format smtp.serveraddress.com
  // but this is not always the case so choosing not to validate the format here
  if (!value) return 'Required';
};

export const validatePort = (value) => {
  if (!value) return 'Required';
};

/**
 * Validating email against a RFC 5322 (https://www.ietf.org/rfc/rfc5322.txt) compliant pattern
 */
export const isValidEmail = (value) =>
  /^(?:[a-z0-9!#$%&'*+\/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+\/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])$/i.test(
    value
  );
