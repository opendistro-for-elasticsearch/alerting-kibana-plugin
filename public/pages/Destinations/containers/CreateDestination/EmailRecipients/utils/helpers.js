/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

export default async function getEmailGroups(searchText = '') {
  const { httpClient } = this.props;
  try {
    const response = await httpClient.get(
      `../api/alerting/email_groups?search=${searchText}&size=200`
    );
    if (response.data.ok) {
      return response.data.emailGroups;
    } else {
      console.log('Unable to get email groups', response.data.err);
      return [];
    }
  } catch (err) {
    console.log('Unable to get email groups', err);
    return [];
  }
}
