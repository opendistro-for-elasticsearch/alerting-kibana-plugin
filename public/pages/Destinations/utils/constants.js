/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

export const DESTINATION_TYPE = {
  CHIME: 'chime',
  SLACK: 'slack',
  CUSTOM_HOOK: 'custom_webhook',
  EMAIL: 'email',
};

export const DESTINATION_OPTIONS = [
  { value: DESTINATION_TYPE.CHIME, text: 'Amazon Chime' },
  { value: DESTINATION_TYPE.SLACK, text: 'Slack' },
  { value: DESTINATION_TYPE.CUSTOM_HOOK, text: 'Custom webhook' },
  { value: DESTINATION_TYPE.EMAIL, text: 'Email' },
];
