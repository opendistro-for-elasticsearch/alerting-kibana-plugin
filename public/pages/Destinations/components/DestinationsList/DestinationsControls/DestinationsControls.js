/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { EuiFieldSearch, EuiFlexGroup, EuiFlexItem, EuiPagination, EuiSelect } from '@elastic/eui';
import { DESTINATION_OPTIONS } from '../../../utils/constants';

const filterTypes = [{ value: 'ALL', text: 'All type' }, ...DESTINATION_OPTIONS];

const propTypes = {
  activePage: PropTypes.number.isRequired,
  pageCount: PropTypes.number.isRequired,
  search: PropTypes.string,
  type: PropTypes.string,
  onSearchChange: PropTypes.func.isRequired,
  onTypeChange: PropTypes.func.isRequired,
  onPageClick: PropTypes.func.isRequired,
  allowList: PropTypes.array.isRequired,
};

const defaultProps = {
  search: '',
  type: 'ALL',
};

const DestinationsControls = ({
  activePage,
  pageCount,
  search,
  type,
  onSearchChange,
  onTypeChange,
  onPageClick,
  allowList,
}) => {
  const allowedOptions = DESTINATION_OPTIONS.filter((option) => allowList.includes(option.value));
  const filterTypes = [{ value: 'ALL', text: 'All type' }, ...allowedOptions];
  return (
    <EuiFlexGroup style={{ padding: '0px 5px' }}>
      <EuiFlexItem>
        <EuiFieldSearch
          fullWidth={true}
          value={search}
          placeholder="Search"
          onChange={onSearchChange}
        />
      </EuiFlexItem>
      <EuiFlexItem grow={false}>
        <EuiSelect options={filterTypes} value={type} onChange={onTypeChange} />
      </EuiFlexItem>
      <EuiFlexItem grow={false} style={{ justifyContent: 'center' }}>
        <EuiPagination pageCount={pageCount} activePage={activePage} onPageClick={onPageClick} />
      </EuiFlexItem>
    </EuiFlexGroup>
  );
};

DestinationsControls.propTypes = propTypes;

DestinationsControls.defaultProps = defaultProps;

export default DestinationsControls;
