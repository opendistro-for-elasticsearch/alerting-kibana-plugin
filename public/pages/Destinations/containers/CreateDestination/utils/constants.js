/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import { DESTINATION_TYPE } from '../../../utils/constants';

export const URL_TYPE = {
  FULL_URL: 'url',
  ATTRIBUTE_URL: 'custom_url',
};

export const CONTENT_TYPE_KEY = 'Content-Type';

const DEFAULT_CONTENT_VALUE = 'application/json';
// TODO:: Change once we have complex forms for the URL like custom webhook
export const formikInitialValues = {
  urlType: 'url',
  name: '',
  type: 'slack',
  [DESTINATION_TYPE.SLACK]: {
    url: '',
  },
  [DESTINATION_TYPE.CHIME]: {
    url: '',
  },
  [DESTINATION_TYPE.CUSTOM_HOOK]: {
    urlType: URL_TYPE.FULL_URL,
    scheme: 'HTTPS',
    headerParams: [
      {
        key: CONTENT_TYPE_KEY,
        value: DEFAULT_CONTENT_VALUE,
      },
    ],
    queryParams: [
      {
        key: '',
        value: '',
      },
    ],
  },
  [DESTINATION_TYPE.MAIL]: {
    host: '',
    port: '25',
    auth: false,
    method: 'plain',
    from: '',
    recipients: '',
    username: '',
    password: '',
  },
};
