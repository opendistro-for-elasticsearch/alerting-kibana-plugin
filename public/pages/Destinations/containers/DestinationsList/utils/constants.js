/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import { DESTINATION_OPTIONS } from '../../../utils/constants';

export const DEFAULT_QUERY_PARAMS = {
  from: 0,
  search: '',
  size: 20,
  sortDirection: 'desc',
  sortField: 'name',
  type: 'ALL',
};

export const MAX_DESTINATIONS = 200;

export const staticColumns = [
  {
    field: 'name',
    name: 'Destination name',
    sortable: true,
    truncateText: true,
    textOnly: true,
    width: '100px',
  },
  {
    field: 'type',
    name: 'Destination type',
    truncateText: true,
    sortable: true,
    textOnly: true,
    width: '100px',
    render: (value) => {
      //TODO:: Convert this to proper map of text to avoid filters always
      const actionType = DESTINATION_OPTIONS.filter((item) => item.value === value);
      if (actionType.length > 0) {
        return actionType[0].text;
      } else {
        return 'Unsupported Type';
      }
    },
  },
  {
    field: 'user',
    name: 'Last updated by',
    sortable: true,
    truncateText: true,
    textOnly: true,
    width: '100px',
    render: (value) => (value && value.name ? value.name : '-'),
  },
];
