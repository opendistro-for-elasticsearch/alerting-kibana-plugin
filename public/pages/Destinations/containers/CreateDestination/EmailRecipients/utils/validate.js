/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import _ from 'lodash';
import { RECIPIENT_TYPE } from './constants';
import { isValidEmail } from '../../../../components/createDestinations/Email/utils/validate';

export const validateEmailRecipients = (options) => {
  if (_.isEmpty(options)) return 'Required';

  let invalidEmails = [];
  for (const option of options) {
    if (option.type === RECIPIENT_TYPE.EMAIL && !isValidEmail(option.value)) {
      invalidEmails.push(option.value);
    }
  }

  if (invalidEmails.length > 0) return `Invalid emails: ${invalidEmails}`;
};
