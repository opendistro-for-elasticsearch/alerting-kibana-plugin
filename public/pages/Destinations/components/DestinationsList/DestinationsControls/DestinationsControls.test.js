/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React from 'react';
import { render, mount } from 'enzyme';
import DestinationsControls from './DestinationsControls';
import { DESTINATION_TYPE } from '../../../utils/constants';

describe('<DestinationsControls />', () => {
  const mockedProps = {
    activePage: 0,
    pageCount: 10,
    onSearchChange: jest.fn(),
    onTypeChange: jest.fn(),
    onPageClick: jest.fn(),
    allowList: Object.values(DESTINATION_TYPE),
  };
  beforeEach(() => jest.resetAllMocks());
  test('should render DestinationsControls', () => {
    const wrapper = render(<DestinationsControls {...mockedProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should invoke handlers with expected values', () => {
    const wrapper = mount(<DestinationsControls {...mockedProps} />);

    // Validate search field SearchText Changes
    const searchText = 'search destination';
    wrapper
      .find('input')
      .at(0)
      .simulate('change', { target: { value: searchText } });
    expect(mockedProps.onSearchChange).toHaveBeenCalledTimes(1);
    const {
      target: { value: searchValue },
    } = mockedProps.onSearchChange.mock.calls[0][0];
    expect(searchValue).toBe(searchText);

    // Validate destination Type Change
    const typeOptionValue = 'chime';
    wrapper.find('select').simulate('change', { target: { value: typeOptionValue } });
    const {
      target: { value: typeValue },
    } = mockedProps.onTypeChange.mock.calls[0][0];
    expect(typeValue).toBe(typeOptionValue);

    // Validate page navigation, Simulate page Click 3
    wrapper.find('button[data-test-subj="pagination-button-3"]').at(0).simulate('click');
    expect(mockedProps.onPageClick).toHaveBeenCalledTimes(1);
    expect(mockedProps.onPageClick.mock.calls[0][0]).toBe(3);
  });
});
