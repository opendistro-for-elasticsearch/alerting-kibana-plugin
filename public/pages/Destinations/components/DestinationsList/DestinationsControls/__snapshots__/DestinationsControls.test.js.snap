// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DestinationsControls /> should render DestinationsControls 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
  style="padding:0px 5px"
>
  <div
    class="euiFlexItem"
  >
    <div
      class="euiFormControlLayout euiFormControlLayout--fullWidth"
    >
      <div
        class="euiFormControlLayout__childrenWrapper"
      >
        <input
          class="euiFieldSearch euiFieldSearch--fullWidth euiFieldSearch-isClearable"
          placeholder="Search"
          type="search"
          value=""
        />
        <div
          class="euiFormControlLayoutIcons"
        >
          <span
            class="euiFormControlLayoutCustomIcon"
          >
            <div>
              EuiIconMock
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiFlexItem euiFlexItem--flexGrowZero"
  >
    <div
      class="euiFormControlLayout"
    >
      <div
        class="euiFormControlLayout__childrenWrapper"
      >
        <select
          class="euiSelect"
        >
          <option
            selected=""
            value="ALL"
          >
            All type
          </option>
          <option
            value="chime"
          >
            Amazon Chime
          </option>
          <option
            value="slack"
          >
            Slack
          </option>
          <option
            value="custom_webhook"
          >
            Custom webhook
          </option>
          <option
            value="email"
          >
            Email
          </option>
        </select>
        <div
          class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
        >
          <span
            class="euiFormControlLayoutCustomIcon"
          >
            <div>
              EuiIconMock
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiFlexItem euiFlexItem--flexGrowZero"
    style="justify-content:center"
  >
    <nav
      class="euiPagination"
    >
      <button
        aria-label="Previous page"
        class="euiButtonIcon euiButtonIcon--text"
        data-test-subj="pagination-button-previous"
        disabled=""
        type="button"
      >
        <div>
          EuiIconMock
        </div>
      </button>
      <ul
        class="euiPagination__list"
      >
        <li
          class="euiPagination__item"
        >
          <button
            aria-current="true"
            aria-label="Page 1 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall euiButtonEmpty-isDisabled euiPaginationButton euiPaginationButton-isActive euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-0"
            disabled=""
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                1
              </span>
            </span>
          </button>
        </li>
        <li
          class="euiPagination__item"
        >
          <button
            aria-label="Page 2 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall euiPaginationButton euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-1"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                2
              </span>
            </span>
          </button>
        </li>
        <li
          class="euiPagination__item"
        >
          <button
            aria-label="Page 3 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall euiPaginationButton euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-2"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                3
              </span>
            </span>
          </button>
        </li>
        <li
          class="euiPagination__item"
        >
          <button
            aria-label="Page 4 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall euiPaginationButton euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-3"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                4
              </span>
            </span>
          </button>
        </li>
        <li
          class="euiPagination__item"
        >
          <button
            aria-label="Page 5 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall euiPaginationButton euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-4"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                5
              </span>
            </span>
          </button>
        </li>
        <li
          aria-label="Skipping pages 6 to 9"
          class="euiPaginationButton-isPlaceholder euiPagination__item"
        >
          â€¦
        </li>
        <li
          class="euiPagination__item"
        >
          <button
            aria-label="Page 10 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall euiPaginationButton euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-9"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                10
              </span>
            </span>
          </button>
        </li>
      </ul>
      <button
        aria-label="Next page, 2"
        class="euiButtonIcon euiButtonIcon--text"
        data-test-subj="pagination-button-next"
        type="button"
      >
        <div>
          EuiIconMock
        </div>
      </button>
    </nav>
  </div>
</div>
`;
